File: 02 백준/10 문자열/Python/Prob4659.py

"""
title  : 4659. 비밀번호 발음하기 (Silver 5)
time   : 40ms        108ms
memory : 31120KB     109108KB
"""

import sys

def solution():

    input = sys.stdin.readline
    
    while True:

        password = input().rstrip()

        if password == 'end':
            break

        if check_aeiou(password) and check_triple(password) and check_sameword(password):
            print("<{}> is acceptable.".format(password))
        else:
            print("<{}> is not acceptable.".format(password))

def check_aeiou(password):

    for word in ['a','e','i','o','u']:
        if word in password:
            return True
    return False

def check_triple(password):

    left_count, right_count = 0, 0

    for word in password:
        
        if word in ['a','e','i','o','u']:
            left_count += 1
            right_count = 0
        else:
            left_count = 0
            right_count += 1
        
        if left_count == 3 or right_count == 3:
            return False
        
    return True

def check_sameword(password):

    idx = 0

    while idx < len(password)-1:

        temp = password[idx]
        idx += 1

        if temp in ['e', 'o']:
            continue

        if temp == password[idx]:
            return False
        
    return True

solution()

File: 02 백준/10 문자열/Java/src/Prob4659.java

/**
* title  : 4659. 비밀번호 발음하기 (Silver 5)
* time   : 88ms
* memory : 14008KB
*/

import java.io.*;

public class Prob4659 {

    private static String vowels = "aeiou";

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        while (true) {

            String word = br.readLine();
            char[] wordArray = word.toCharArray();

            if (word.equals("end")) {
                break;
            }

            sb.append("<").append(word).append("> is ");

            if (checkOne(wordArray) && checkTwo(wordArray) && checkThree(wordArray)) {
                sb.append("acceptable.").append("\n");
            } else {
                sb.append("not acceptable.").append("\n");
            }
        }

        System.out.println(sb);
    }

    public static boolean checkOne(char[] word) {

        for (int i = 0; i < word.length; i++) {
            if (vowels.contains(String.valueOf(word[i]))) {
                return true;
            } 
        }

        return false;
    }

    public static boolean checkTwo(char[] word) {

        int vCount = 0;
        int eCount = 0;

        for (int i = 0; i < word.length; i++) {

            if (vowels.contains(String.valueOf(word[i]))) {
                vCount++;
                eCount = 0;
            } else {
                eCount++;
                vCount = 0;
            }

            if (vCount == 3 || eCount == 3) {
                return false;
            }
        }

        return true;
    }

    public static boolean checkThree(char[] word) {

        for (int i = 1; i < word.length; i++) {

            if (word[i-1] == word[i]) {
                if (word[i] == 'e' || word[i] == 'o') {
                    continue;
                }
                return false;
            }
        }

        return true;
    }
}